airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2109,3 +2122,287 @@ def execute(self, context: Any):
+            raise AirflowException(f"argument 'validation_type' must be one of {great_expectations_valid_type}")
+        self.validation_type = validation_type
+        self.validation_type_input = validation_type_input
+        self.bigquery_conn_id = bigquery_conn_id
+        self.bq_dataset_name = bq_dataset_name
+        self.gcp_project = gcp_project
+        self.gcs_bucket = gcs_bucket
+        self.gcs_expectations_prefix = gcs_expectations_prefix
+        self.gcs_validations_prefix = gcs_validations_prefix
+        self.gcs_datadocs_prefix = gcs_datadocs_prefix
+        self.email_to = email_to
+        self.datadocs_domain = datadocs_domain
+        self.send_alert_email = send_alert_email
+        self.datadocs_link_in_email = datadocs_link_in_email
+        self.fail_if_expectations_not_met = fail_if_expectations_not_met
+
+    def instantiate_data_context(self):
+
+        conn = BaseHook.get_connection(self.bigquery_conn_id)
+        connection_json = conn.extra_dejson
+
+        project_config = DataContextConfig(
+            config_version=2,
+            datasources={
+                "bq_datasource": {
+                    "credentials": {
+                        "url": "bigquery://" + connection_json[
+                            'extra__google_cloud_platform__project'] + "/" + self.bq_dataset_name + "?credentials_path=" +
+                               connection_json['extra__google_cloud_platform__key_path']
+                    },
+                    "class_name": "SqlAlchemyDatasource",
+                    "module_name": "great_expectations.datasource",
+                    "data_asset_type": {
+                        "module_name": "great_expectations.dataset",
+                        "class_name": "SqlAlchemyDataset"
+                    }
+                }
+            },
+            expectations_store_name="expectations_GCS_store",
+            validations_store_name="validations_GCS_store",
+            evaluation_parameter_store_name="evaluation_parameter_store",
+            plugins_directory=None,
+            validation_operators={
+                "action_list_operator": {
+                    "class_name": "ActionListValidationOperator",
+                    "action_list": [
+                        {
+                            "name": "store_validation_result",
+                            "action": {"class_name": "StoreValidationResultAction"},
+                        },
+                        {
+                            "name": "store_evaluation_params",
+                            "action": {"class_name": "StoreEvaluationParametersAction"},
+                        },
+                        {
+                            "name": "update_data_docs",
+                            "action": {"class_name": "UpdateDataDocsAction"},
+                        },
+                    ],
+                }
+            },
+            stores={
+                'expectations_GCS_store': {
```

Comment:
I feel like setting the expectations store and the validations store (or all of the data context at that) should be abstracted away from here somehow, perhaps by creating a great expectations hook. That would allow to, say, use a gcs bucket for these stores and create a snowflake great expectations operator perhaps.
```python
datasource = ge_gcs_hook.create_bigquery_datasource(bigquery_conn_id, bq_dataset_name, other_params) 
data_context = ge_gcs_hook.create_data_context(datasource, gcs_conn_id, gcs_bucket, gcs_validations_prefix, gcs_expectations_prefix)
```
so we can also have something like
```python
# Base hook sets abstract on data context
custom_datasource = ge_hook.create_custom_datasource(conn_id, dataset_name, class_name, module_name) 
```
Something like that